server/src/test/java/org/elasticsearch/rest/RestRequestTests.java

```diff
@@ -118,6 +118,58 @@ public void testApplyContentParser() throws IOException {
         assertEquals(emptyMap(), source.get());
     }
 
+    public void testParamAsIntWithNoParameters() {
+        RestRequest restRequest = contentRestRequest("", emptyMap());
+        int defaultValue = randomInt();
+        String parameterKey = randomIdentifier();
+
+        int value = restRequest.paramAsInt(parameterKey, defaultValue);
+        assertEquals(defaultValue, value);
+    }
+
+    public void testParamAsIntWithIntegerParameter() {
+        String parameterKey = randomIdentifier();
+        RestRequest restRequest = contentRestRequest("", singletonMap(parameterKey, "123"));
+        int defaultValue = randomInt();
+
+        int value = restRequest.paramAsInt(parameterKey, defaultValue);
+        assertEquals(123, value);
+    }
+
+    public void testParamAsIntWithoutIntegerParameter() {
```

Comment:
naming nit: "without" suggests the parameter is absent, but in fact it's present, it's just that its value is not numeric (likewise below)

```suggestion
    public void testParamAsIntWithNonIntegerParameter() {
```

Reply:
Fixed in e87058d46512036865c5f97f66ae755eca1014ad